From f32eb7c35806ecb0c3e1c2c78b93a30e067ad889 Mon Sep 17 00:00:00 2001
Message-Id: <f32eb7c35806ecb0c3e1c2c78b93a30e067ad889.1429601425.git.chang-joon.lee@intel.com>
In-Reply-To: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
References: <20df33cbe9e95d11653023b1a6201653c517d353.1429601425.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Sat, 11 Apr 2015 00:21:01 +0530
Subject: [PATCH 5/9] SQUASHME! [VPG]: drm/i915: Add resolution check for
 enabling ddrdvfs.

This patch is to enable ddr dvfs only if the resolution
is less than 19x12 as suggested by SV.

SQUASHME: This patch needs to be squashed with
	commit e8c876f7944d03b2748c9afe355ba57d6d313e25
	Author: vandita kulkarni <vandita.kulkarni@intel.com>
	Date:   Fri Apr 10 21:10:24 2015 +0530
		REVERTME: [VPG]: drm/i915 : program dynamic dsparb and wm

Issue: GMINL-4468
Change-Id: I04319b8266b5cc3943ff619d1db9b478f72e67eb
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |   31 ++++++++++++++++++++++++-------
 1 file changed, 24 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 0b72623..2c726f8 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1730,6 +1730,8 @@ void intel_update_maxfifo(struct drm_i915_private *dev_priv,
 				struct drm_crtc *crtc, bool enable)
 {
 	unsigned int val = 0;
+	struct intel_crtc_config *config =
+				&((to_intel_crtc(crtc))->config);
 
 	if (!IS_VALLEYVIEW(dev_priv->dev))
 		return;
@@ -1737,9 +1739,16 @@ void intel_update_maxfifo(struct drm_i915_private *dev_priv,
 	if (enable) {
 		if (IS_CHERRYVIEW(dev_priv->dev)) {
 			val = 0x0;
-			mutex_lock(&dev_priv->rps.hw_lock);
-			vlv_punit_write(dev_priv, CHV_DDR_DVFS, val);
-			mutex_unlock(&dev_priv->rps.hw_lock);
+			/*
+			 * cannot enable ddr dvfs if
+			 * resolution greater than 19x12
+			 */
+			if ((config->pipe_src_w * config->pipe_src_h)
+					<= (1920 * 1200)) {
+				mutex_lock(&dev_priv->rps.hw_lock);
+				vlv_punit_write(dev_priv, CHV_DDR_DVFS, val);
+				mutex_unlock(&dev_priv->rps.hw_lock);
+			}
 			I915_WRITE(FW_BLC_SELF_VLV, FW_CSPWRDWNEN);
 			mutex_lock(&dev_priv->rps.hw_lock);
 			val = vlv_punit_read(dev_priv, CHV_DPASSC);
@@ -1751,10 +1760,18 @@ void intel_update_maxfifo(struct drm_i915_private *dev_priv,
 		dev_priv->maxfifo_enabled = true;
 	} else {
 		if (IS_CHERRYVIEW(dev_priv->dev)) {
-			val = CHV_FORCE_DDR_HIGH_FREQ |	CHV_DDR_DVFS_DOORBELL;
-			mutex_lock(&dev_priv->rps.hw_lock);
-			vlv_punit_write(dev_priv, CHV_DDR_DVFS, val);
-			mutex_unlock(&dev_priv->rps.hw_lock);
+			/*
+			 * cannot enable ddr dvfs if
+			 * resolution is greater than 19x12
+			 */
+			if ((config->pipe_src_w * config->pipe_src_h)
+					<= (1920 * 1200)) {
+				val = CHV_FORCE_DDR_HIGH_FREQ |
+						CHV_DDR_DVFS_DOORBELL;
+				mutex_lock(&dev_priv->rps.hw_lock);
+				vlv_punit_write(dev_priv, CHV_DDR_DVFS, val);
+				mutex_unlock(&dev_priv->rps.hw_lock);
+			}
 			mutex_lock(&dev_priv->rps.hw_lock);
 			val = vlv_punit_read(dev_priv, CHV_DPASSC);
 			I915_WRITE(FW_BLC_SELF_VLV, ~FW_CSPWRDWNEN);
-- 
1.7.9.5

